src/v0/destinations/mp/util.js

```diff
@@ -166,6 +170,102 @@ const removeDuplicateMetadata = (mergedBatches) => {
   }
 };
 
+/**
+ * Groups events with the same message type together in batches.
+ * Each batch contains events that have the same message type and are from different users.
+ * @param {*} inputs - An array of events
+ * @returns {*} - An array of batches
+ */
+const groupEventsByType = (inputs) => {
+  const batches = [];
+  let currentInputsArray = inputs;
+  while (currentInputsArray.length > 0) {
+    const remainingInputsArray = [];
+    const userOrderTracker = {};
+    const event = currentInputsArray.shift();
+    const messageType = event.message.type;
+    const batch = [event];
+    currentInputsArray.forEach((currentInput) => {
+      const currentMessageType = currentInput.message.type;
+      const currentUser = currentInput.metadata.userId;
+      if (currentMessageType === messageType && !userOrderTracker[currentUser]) {
+        batch.push(currentInput);
+      } else {
+        remainingInputsArray.push(currentInput);
+        userOrderTracker[currentUser] = true;
+      }
+    });
+    batches.push(batch);
+    currentInputsArray = remainingInputsArray;
+  }
+
+  return batches;
+};
+
+/**
+ * Group events with the same endpoint together in batches
+ * @param {*} events - An array of events
+ * @returns
+ */
+const groupEventsByEndpoint = (events) => {
+  const engageEvents = [];
+  const groupsEvents = [];
+  const trackEvents = [];
+  const importEvents = [];
+  const batchErrorRespList = [];
+  events.forEach((result) => {
+    if (result.message) {
+      const { destination, metadata } = result;
+      let { message } = result;
+      message = CommonUtils.toArray(message);
+      message.forEach((msg) => {
+        // eslint-disable-next-line default-case
+        switch (true) {
```

Comment:
Isn't there a better way to do this?

Reply:
refactor the code and replaced the switch case and used .find()